### PR TITLE
ci/lava: allow resubmitting LAVA Test failures

### PR DESCRIPTION
**Pull request by mwasilew**
This patch broadens the list of potential test job error messages that
qualify for automated resubmission. Also messages can now be added to
LAVA ci backend setting as regular expressions.

Signed-off-by: Milosz Wasilewski <milosz.wasilewski@linaro.org>

### CODE REVIEW

File: squad/ci/backend/lava.py [hunk 1]
```diff
@@ -327,8 +328,10 @@ def __parse_results__(self, data, test_job):
                         if metadata['error_type'] in ['Infrastructure', 'Lava', 'Job']:
                             completed = False
                         # automatically resubmit in some cases
-                        if metadata['error_type'] in ['Infrastructure', 'Job'] and \
-                                any(substring.lower() in metadata['error_msg'].lower() for substring in lava_infra_error_messages):
+                        infra_messages_re = re.compile("|".join(lava_infra_error_messages))
+                        if metadata['error_type'] in ['Infrastructure', 'Job', 'Test'] and \
+                                len(lava_infra_error_messages) > 0 and \
+                                infra_messages_re.search(metadata['error_msg']) is not None:
```
Comment (terceiro) [inline squad/ci/backend/lava.py]: are you dropping the case insensitivity (`.lower()`) on purpose?

Comment (terceiro) [reply squad/ci/backend/lava.py]: you should probably handle invalid regular expressions

Comment (terceiro) [reply squad/ci/backend/lava.py]: I'm in doubt wrt joining all the items like with (`"|".join(...)`). if one of the items is not a valid regular expression, you will invalidate the entire check. I would keep testing one by one, and ignoring the ones that are not valid regexes.

**Comment by mwasilew** [comment]
I added re.I so the regex won't be case sensitive. Also added a check for incorrect expressions which are ignored. 

I had issues saving \d (or similar) strings into YAML in the tests. Not sure if that works when added in the admin interface.

**Pull request merged**